### PR TITLE
Support parse UUID without separator

### PR DESCRIPTION
**Pull request by zhang2014**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Support parse UUID without separator(separators are always removed in most implementations, this is helpful for users to write data)

### CODE REVIEW

File: src/IO/ReadHelpers.h [hunk 1]
```diff
@@ -577,13 +636,18 @@ inline void readUUIDText(UUID & uuid, ReadBuffer & buf)
     char s[36];
     size_t size = buf.read(s, 36);
```
Comment (alexey-milovidov) [inline src/IO/ReadHelpers.h:637]: But how we will continue to read next data from buffer if we already read extra bytes?

File: src/IO/ReadHelpers.h [hunk 2]
```diff
@@ -487,8 +488,66 @@ struct NullSink
...
-void parseUUID(const UInt8 * src36, UInt8 * dst16);
-void parseUUID(const UInt8 * src36, std::reverse_iterator<UInt8 *> dst16);
+template <typename IteratorSrc, typename IteratorDst>
+void parseHex(IteratorSrc src, IteratorDst dst, const size_t num_bytes)
+{
+    size_t src_pos = 0;
+    size_t dst_pos = 0;
+    for (; dst_pos < num_bytes; ++dst_pos)
+    {
+        dst[dst_pos] = UInt8(unhex(src[src_pos])) * 16 + UInt8(unhex(src[src_pos + 1]));
```
Comment (alexey-milovidov) [inline src/IO/ReadHelpers.h:498]: The `unhex2` function is exactly for this purpose.

File: src/IO/ReadHelpers.h [hunk 3]
```diff
@@ -487,8 +488,66 @@ struct NullSink
...
+        src_pos += 2;
+    }
+}
+
+template <bool with_separator>
+void parseUUID(const UInt8 * src36, UInt8 * dst16)
```
Comment (alexey-milovidov) [inline src/IO/ReadHelpers.h:504]: This template is always used with explicit parameter. And it is implemented with two separate branches...

So, we can simply make two functions:

parseUUID
parseUUIDWithoutSeparator

File: src/IO/ReadHelpers.h [hunk 4]
```diff
@@ -487,8 +488,66 @@ struct NullSink
...
+    {
+        parseHex(&src36[0], &dst16[0], 4);
+        parseHex(&src36[9], &dst16[4], 2);
+        parseHex(&src36[14], &dst16[6], 2);
+        parseHex(&src36[19], &dst16[8], 2);
+        parseHex(&src36[24], &dst16[10], 6);
+    }
+    else
+    {
+        parseHex(&src36[0], &dst16[0], 4);
```
Comment (alexey-milovidov) [inline src/IO/ReadHelpers.h:517]: Why don't parse it all with one call of `parseHex`?

**Comment by qoega** [comment]
Better to add examples to docs about UUID

**Pull request merged**